### PR TITLE
Mobile View

### PR DESCRIPTION
**Pull request by jasonkalmeida**
https://github.com/techforwarren/eventmap/issues/15

Utilizing `react-device-detect`, we can now detect device types.

Using that new functionality, I've created a mobile specific component for the events - a card/carousel view.

Additionally, I've put conditional renders on components that don't make sense on mobile.

(Working to get this on my own fork so I can host it to view - in the meantime if you could localhost)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -23,13 +27,25 @@ function App() {
...
       setLocFilt(null);
+      setCardIndex(0);
 
     }
   }, [currZip]);
 
+  //Card index utilizes the hoverEvent to highlight the card's respective marker
+  useEffect(() => {
+    if(isMobile && events != null){
+      setHoverEvent((('location' in events[cardIndex] && 'location' in events[cardIndex]['location'] && 'latitude' in events[cardIndex]['location']['location']) ? "" + events[cardIndex]['location']['location']['latitude'] + "&" + events[cardIndex]['location']['location']['longitude'] : null));
```
Comment (joegoldbeck) [inline src/App.js]: Nit: this line is silly long. Maybe consider a linter and/or auto-formatter to take care of that stuff

File: src/EventList.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ export function EventList(props) {
     //Location filter
     if(props.locFilt != null){
       if('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']){
-        if(event['location']['location']['latitude'] !== props.locFilt['lat'] || event['location']['location']['longitude'] != props.locFilt['lng']){
+        if(event['location']['location']['latitude'] !== props.locFilt['lat'] || event['location']['location']['longitude'] !== props.locFilt['lng']){
```
Comment (joegoldbeck) [inline src/EventList.js]: This event[location][location] is a bit wordy. Maybe consider processing the event upon arrival and either making it an instance of some ‘event’ class (would probably me cleanest), or otherwise simplify the object for easier access of the properties you tend to access.

File: src/EventList.js [hunk 3]
```diff
@@ -74,12 +74,13 @@ export function EventList(props) {
...
-        target="_blank" 
-        key={event['id']} 
-        coord={('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) ? "" + event['location']['location']['latitude'] + "&" + event['location']['location']['longitude'] : ""} 
-        onMouseEnter={(event) => { props.updatedHover(event['currentTarget'].getAttribute('coord')) }} 
+      <a href={event['browser_url']}
+        className="eventCard"
+        target="_blank"
+        rel="noopener noreferrer"
+        key={event['id']}
+        coord={('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) ? "" + event['location']['location']['latitude'] + "&" + event['location']['location']['longitude'] : ""}
```
Comment (joegoldbeck) [inline src/EventList.js]: Yeah, there are a ton of these. It’s hard to grok exactly what this long list of conditionals is checking. 

A class would really clean a lot of this up, or alternatively, a function that is evocatively named and accepts the event as an arg to replace these in-line conditionals

File: src/MobileList.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
+import React, { useEffect } from 'react';
+import moment from 'moment';
+import groupBy from 'lodash.groupby';
+import sortBy from 'lodash.sortby';
+require('twix');
```
Comment (joegoldbeck) [inline src/MobileList.js]: Is this rando ‘require’ intentional?

File: src/MobileList.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+const MAX_DAYS_IN_LIST = 1;
+
+function EventTimes(props) {
+  const { rawTimes } = props;
+  let sortedTimesByDate = groupBy(sortBy(rawTimes,
```
Comment (joegoldbeck) [inline src/MobileList.js]: Are you familiar with the chained syntax for lodash? Would make this cleaner to read since the functions would be in order instead of nested

File: src/MobileList.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+          <p><strong>{event['location']['venue']}</strong> in <strong>{event['location']['locality']}</strong></p>
+          <EventTimes rawTimes={rawTimes} />
+          <p className="eventRSVP">Click to RSVP</p>
+        </div>
+
+      </a>
+
+    )
+  }).filter((arrItem) => {
+    return arrItem != null;
```
Comment (joegoldbeck) [inline src/MobileList.js]: There are a lot of != null checks going on. 

That always worries me because JS vars tend to be undefined occasionally. Is there an unintentional distinction being made between null and undefined? Would _.isNil work, or are you relying on the weird behavior of != vs !==? The double vs single inequality is always a bit confusing to me and seems to make typo bugs easier

File: src/App.js [hunk 7]
```diff
@@ -23,13 +27,25 @@ function App() {
...
+    }
+  }, [cardIndex]);
+
+
   return (
     <div className="app">
       <SearchBar currZip={currZip} updateZip={(newZip) => setCurrZip(newZip)} events={events} updatedHover={(newHover) => setHoverEvent(newHover)} locFilt={locFilt}/>
+      {events !== null && isMobile &&
+        <MobileList events={events} updatedHover={(newHover) => setHoverEvent(newHover)} locFilt={locFilt} cardIndex={cardIndex} updateCardIndex={(update) => setCardIndex(update)}/>
```
Comment (joegoldbeck) [inline src/App.js]: Are the lambdas serving a purpose here? (i.e. making sure only a single arg gets passed in?) Or could this more simply be ‘updatedHover=setHoverEvent’

File: src/MobileList.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
+import React, { useEffect } from 'react';
+import moment from 'moment';
+import groupBy from 'lodash.groupby';
+import sortBy from 'lodash.sortby';
+require('twix');
+
+const MAX_DAYS_IN_LIST = 1;
+
+function EventTimes(props) {
+  const { rawTimes } = props;
```
Comment (joegoldbeck) [inline src/MobileList.js]: Could maybe use a quick comment about what this fn does

File: src/MobileList.js [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    return sortedDates.slice(0,MAX_DAYS_IN_LIST - 1).map(dateRowFactory).concat(
+      <p>
+        More Times from {nextStart.twix(lastStart, { allDay : true }).format()}
+      </p>
+    )
+  }
+}
+
+export function MobileList(props){
+
```
Comment (joegoldbeck) [inline src/MobileList.js]: Could maybe use a quick comment about what this function does. 

I take it this generates jsx for a list of events to be used in the mobile view?

File: src/EventList.js [hunk 10]
```diff
@@ -74,12 +74,13 @@ export function EventList(props) {
...
-      <a href={event['browser_url']} 
-        className="eventCard" 
-        target="_blank" 
-        key={event['id']} 
-        coord={('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) ? "" + event['location']['location']['latitude'] + "&" + event['location']['location']['longitude'] : ""} 
-        onMouseEnter={(event) => { props.updatedHover(event['currentTarget'].getAttribute('coord')) }} 
+      <a href={event['browser_url']}
+        className="eventCard"
+        target="_blank"
+        rel="noopener noreferrer"
```
Comment (joegoldbeck) [inline src/EventList.js]: Why noreferrer? Would we not want the ewarren mobilize site to know these inbounds are coming from this event map?

**Comment by joegoldbeck** [comment]
Some weird zoom behavior: when I accidentally finger zoom on the event text, instead of the map section, the whole page zooms. It’s kind of weird. Would it be possible to lock the whole page zoom, so only map zoom was a thing?

**Comment by joegoldbeck** [comment]
It’s still a little jarring to show up to an empty map, and then realize I have to enter zip code up top. Absent a fancier solution, would it be possible to bring “enter your zip code” to the middle (either the search box, or just an overlay with text) until you do that initial search?

There’s just a quick moment of “uhhh, why am I staring at an empty map” that I experience every time I do the initial load

File: src/MobileList.js (hunk 4 above)
Comment (jasonkalmeida) [reply src/MobileList.js]: cc: @john-osullivan 

Hey John, can you review this?

File: src/MobileList.js (hunk 5 above)
Comment (jasonkalmeida) [reply src/MobileList.js]: cc: @john-osullivan 

Another item here, John, if you could just speak to this.
